sympy/sets/conditionset.py

```diff
@@ -34,6 +34,8 @@ def __new__(cls, sym, condition, base_set):
             return S.EmptySet
         if condition == S.true:
             return base_set
+        if getattr(base_set, "is_FiniteSet", False) and base_set.is_FiniteSet:
```

Comment:
The test is good and the modified pretty tests are fine.

For the getattr/isinstance issue:  you can use `isinstance(base_set, (FiniteSet, EmptySet))` which makes it clearer that EmptySet is being handled.